File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/NotifySubscribeSuccessBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class NotifySubscribeSuccessBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    companion object {
        private const val notifySubscribeSuccessErrorCode =
            "usercentrics_flutter_notifySubscribeSuccess_error"
    }

    override val name: String
        get() = "notifySubscribeSuccess"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        usercentrics.instance.notifySubscribeSuccess(
            onSuccess = {
                result.success(null)
            },
            onError = {
                result.error(
                    notifySubscribeSuccessErrorCode,
                    it.message,
                    it
                )
            },
        )
    }
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/NotifySubscribeSuccessBridge.swift
import Flutter
import Foundation

struct NotifySubscribeSuccessBridge : MethodBridge {

    let name: String = "notifySubscribeSuccess"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)

        usercentrics.shared.notifySubscribeSuccess(onSuccess: {
            result(nil)
        }, onError: { error in
            result(FlutterError(code: "usercentrics_flutter_notifySubscribeSuccess_error",
                                message: error.localizedDescription,
                                details: nil))
        })
    }
}
